packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFile.kt

```diff
@@ -58,4 +59,15 @@ class FileSystemFile(path: File) : FileSystemPath(path) {
     validateType()
     return path.readText()
   }
+
+  @OptIn(ExperimentalStdlibApi::class)
+  val md5: String get() {
+    val md = MessageDigest.getInstance("MD5")
+    val digest = md.digest(path.readBytes())
```

Comment:
Good catch, but I think I'll keep the error handling in the module file to keep it consistent

Reply:
Of course - just make sure it is handled so that we return null in the correct cases :)

Reply:
<img width="417" alt="image" src="https://github.com/user-attachments/assets/345a70a9-708c-4e0d-a7ed-4917af8311d6">

Almost 😅 

It seems that returning null from a Property is actually `undefined` in JS on Android – @lukmccall any ideas on how to actually return null from native?

Reply:
It looks like setting a property to `null` does not actually set the property to the value `null` but it calls `unsetProperty` on it - which basically sets the value to `undefined`:
(https://github.com/expo/expo/blob/fa26ba89434ad47e06195c36b5a8f20f013b995e/packages/expo-modules-core/android/src/main/cpp/JavaScriptObject.cpp#L137)

With this behaviour maybe we could change the API so that `size`/`md5` returns undefined if file is not found?

Reply:
Yeah, I think that's the codepath for the testcase I shared, but it's not directly related to returning `null` from `Property` – I've tested and `unsetProperty` is not called when that happens.

I'd rather keep null as return – seems conceptually cleaner to me, and we can always wrap it in JS land, mostly wondering about this behaviour in core – wdyt about it?

Reply:
Found the place to do that in [`39b5612` (#31301)](https://github.com/expo/expo/pull/31301/commits/39b5612f432d68fde6e8eafef13f2a726ecc334b) – let's see what @lukmccall says :)

Reply:
Sounds reasonable - then we'll wait a bit for @lukmccall to respond before finalising the review maybe.

Reply:
That’s a bit tricky. You’ve found the correct place. However, it was coded like this to ensure parity with iOS. If the behavior is different between platforms, we can change it. So, if the iOS returns `null` in those cases, feel free to change the converter.

Reply:
Changing the return value should maybe considered being a breaking change - consumers like other libraries might do tests like `value === undefined` which will fail after such a change - Not sure about how widespread this problem could be?

Reply:
@lukmccall iOS returns null, but I agree this is a breaking change. Not too widespread imo.

Reply:
On one platform, checking `value === undefined` will fail, while on another, it would work. Therefore, I consider it a bug fix. Let's change the core to always return `null`

Reply:
Looks like we might be ready to approve this one for now, then? @aleqsio @lukmccall :)